min_queue.c

```diff
@@ -59,8 +59,10 @@ struct minqueue {
 minqueue_t *minq_new(minq_key_cmp cmp)
 {
 	minqueue_t *q;
-	if (!cmp || !(q = malloc(sizeof(*q))))
+	if (!cmp || !(q = malloc(sizeof(*q)))) {
+		free(q);
```

Comment:
I don't think this change is right. Either there is no cmp (and then no malloc is done) or the malloc fails (and no free is required)

Reply:
Good point. Actually I found it with [cppcheck](http://cppcheck.sourceforge.net), who clearly indicates a memory leak here. Double checking it here proves that we indeed do not need to free memory. Thank you for pointing that out.

Reply:
Static analyzers sometimes produce false positives :-)